Update-HostsFile.ps1

```diff
@@ -0,0 +1,170 @@
+function Update-HostsFile
+{
+    <#
+    .SYNOPSIS
+    This CmdLet downloads the unified hosts file with optional extensions from 
+    https://github.com/StevenBlack/hosts and copies it to the correct folder on Windows machines. 
+
+    .PARAMETER UnifiedOnly
+    Downloads only the base hosts file without extensions
+
+    .PARAMETER Fakenews
+    Downloads the base hosts file with the Fakenews extension
+
+    .PARAMETER Gambling
+    Downloads the base hosts file with the Gambling extension
+
+    .PARAMETER Social
+    Downloads the base hosts file with the Social extension
+
+    .PARAMETER Porn
+    Downloads the base hosts file with the Porn extension
+
+    .PARAMETER All
+    Downloads the base hosts file with all extensions
+
+    .PARAMETER WhitelistFilePath
+    Path to a file which contains a list with domains which need to be whitelisted. The domains will
+    be cut out of the hosts file. 
+    The file must only contain the domains seperated be new line.
+
+    .PARAMETER BlackListFilePath
+    Path to a file which contains a list with domains which need to be blacklisted. The domains will
+    be appended to the end of the hosts file. 
+    The file must only contain the domains seperated be new line.
+
+    .PARAMETER BackupOldHostsFile
+    Tells the function to backup the original hosts file.
+
+    .EXAMPLE
+    C:\PS> Update-HostsFile
+    Downloads the file without extensions and writes the content into 
+    %SYSTEMROOT%\System32\etc\drivers\hosts
+
+    .EXAMPLE
+    C:\PS> Update-HostsFile -FakeNews -Social
+    Downloads the file with the social and Fakenews extensions and writes the content into 
+    %SYSTEMROOT%\System32\etc\drivers\hosts. You can combine any of the extensions.
+
+    .EXAMPLE
+    C:\PS> Update-HostsFile -Gambling -Social -WhiteListFilePath \\share\whitelist.txt 
+    Downloads the file with the social and Gambling extensions, cuts out the whitelisted domains

```

Comment:
"social" lowercase but "gambling" uppercase? (Multiple instances where you do this).